nats.go

```diff
@@ -1208,6 +1227,16 @@ func (nc *Conn) checkForSecure() error {
 		o.Secure = true
 	}
 
+	if o.Nkey != "" && nc.info.Nonce == "" {
```

Comment:
So if user defines NKey, then it can connect only to servers that support  it, is that ok?

Reply:
I believe so, if a user defines them that is what they expect and if the server does not support it that should error IMO.